src/test/java/jp/ikedam/jenkins/plugins/updatesitesmanager/DescribedUpdateSiteSimpleTest.java

```diff
@@ -23,205 +23,93 @@
-        
-        // null
-        {
-            new TestDescribedUpdateSite(null, null);
-        }
-    }
-    
-    public void testGetUpdateSite()
-    {
-        DescribedUpdateSite site = new TestDescribedUpdateSite("id", "url");
-        assertSame("getUpdateSite must return itself", site, site.getUpdateSite());
-    }
-    
-    /*
-    public void testGetAvailables()
-    {
-        // no way to initialize UpdateSite...
-    }
-    */
-    
-    /*
-    public void testGetUpdates()
-    {
-        // no way to initialize UpdateSite...
-    }
-    */
-    
-    /*
-    public void testIsDue()
-    {
-        // no way to initialize UpdateSite...
+
+    @Test
+    public void shouldTrimUrlAndId() {
+        String id = "testId";
+        String url = "http://localhost/update-center.json";
+        UpdateSite site = new TestDescribedUpdateSite(
+                "  " + id + "  ",
+                " " + url + "\t"
+        );
+
+        assertThat("id is not trimmed", site.getId(), is(id));
+        assertThat("url is not trimmed", site.getUrl(), is(url));
     }
-    */
-    
-    /*
-    public void testHasUpdates()
-    {
-        // no way to initialize UpdateSite...
+
+    @Test
+    public void shoulHandleNull() {
+        new TestDescribedUpdateSite(null, null);
     }
-    */
-    
-    private DescribedUpdateSite.Descriptor getDescriptor()
-    {
+
+    private DescribedUpdateSite.Descriptor getDescriptor() {
         return new TestDescribedUpdateSite.DescriptorImpl();
     }
-    
-    public void testDescriptorDoCheckIdOk()
-    {
-        DescribedUpdateSite.Descriptor descriptor = getDescriptor();
-        
-        {
-            String id = "somevalue";
-            assertEquals(
-                "simple id",
-                FormValidation.Kind.OK,
-                descriptor.doCheckId(id).kind
-            );
-        }
+
+    @Test
+    public void testDescriptorDoCheckIdOk() {
+        assertThat("simple id", getDescriptor().doCheckId("somevalue").kind, is(OK));
     }
-    
-    public void testDescriptorDoCheckIdError()
-    {
-        DescribedUpdateSite.Descriptor descriptor = getDescriptor();
-        
-        // null
-        {
-            String id = null;
-            assertEquals(
-                "null",
-                FormValidation.Kind.ERROR,
-                descriptor.doCheckId(id).kind
-            );
-        }
-        
-        // empty
-        {
-            String id = "";
-            assertEquals(
-                "empty",
-                FormValidation.Kind.ERROR,
-                descriptor.doCheckId(id).kind
-            );
-        }
-        
-        // blank
-        {
-            String id = "  ";
-            assertEquals(
-                "blank",
-                FormValidation.Kind.ERROR,
-                descriptor.doCheckId(id).kind
-            );
-        }
+
+    @Test
+    @DataProvider(value = {
+            "empty,",
+            "blank, ",
+            "null value,null"

```

Comment:
(Note for me)
`DataProvider` converts "null" to `null`.
https://github.com/TNG/junit-dataprovider/wiki/Features